pkg/controllers/backup/cluster_origin.go

```diff
@@ -0,0 +1,109 @@
+package backup
+
+import (
+	v1 "github.com/rancher/backup-restore-operator/pkg/apis/resources.cattle.io/v1"
+	"github.com/rancher/wrangler/v3/pkg/condition"
+)
+
+type backupClusterOriginConditionMeta struct {
+	backupName                     string
+	hasClusterOriginID             bool
+	clusterOriginID                string
+	hasCurrentOriginCondition      bool
+	currentOriginCondition         bool
+	canInPlaceRestore              bool
+	hasInPlaceRestoreCondition     bool
+	currentInPlaceRestoreCondition bool
+}
+
+func newBackupClusterOriginConditionMeta(controllerClusterID string, backup *v1.Backup) backupClusterOriginConditionMeta {
+	conditionMeta := backupClusterOriginConditionMeta{
+		backupName:                     backup.Name,
+		hasClusterOriginID:             false,
+		hasCurrentOriginCondition:      false,
+		currentOriginCondition:         false,
+		canInPlaceRestore:              false,
+		hasInPlaceRestoreCondition:     false,
+		currentInPlaceRestoreCondition: false,
+	}
+
+	originalValue := backup.Status.OriginCluster
+	conditionMeta.hasClusterOriginID = originalValue != ""
+	if conditionMeta.hasClusterOriginID {
+		conditionMeta.clusterOriginID = originalValue
+	}
+
+	currentOriginConditionString := condition.Cond(v1.BackupConditionClusterOrigin).GetStatus(backup)
+	conditionMeta.hasCurrentOriginCondition = currentOriginConditionString != ""
+	if !conditionMeta.hasCurrentOriginCondition {
+		conditionMeta.currentOriginCondition = currentOriginConditionString == "True"
+	}
+
+	if conditionMeta.hasClusterOriginID {
+		conditionMeta.canInPlaceRestore = conditionMeta.clusterOriginID == controllerClusterID
+	}
+
+	currentInPlaceRestoreString := condition.Cond(v1.BackupConditionInPlaceRestore).GetStatus(backup)
+	conditionMeta.hasInPlaceRestoreCondition = currentInPlaceRestoreString != ""
+	if !conditionMeta.hasInPlaceRestoreCondition {
+		conditionMeta.currentInPlaceRestoreCondition = currentInPlaceRestoreString == "True"
+	}
+
+	return conditionMeta
+}
+
+// prepareClusterOriginConditions helps set the cluster origin conditions and reports if anything changed in this part of status.
+func (h *handler) prepareClusterOriginConditions(backup *v1.Backup) bool {
```

Comment:
I think a lot of this logic in this function can also be simplified/more readable if we encapsulate most of it in struct methods